ntbs-service/DataAccess/UserRepository.cs

```diff
@@ -176,22 +182,17 @@ private static void SyncCaseManagerTbServices(User user, IEnumerable<TBService>
             }
         }
 
-        private static void RemoveUnmatchedCaseManagerTbServices(
+        private void RemoveUnmatchedCaseManagerTbServices(
             ICollection<CaseManagerTbService> userCaseManagerTbServices,
             IList<CaseManagerTbService> adCaseManagerTbServices)
         {
-            var caseManagerTbServicesToRemove = new List<CaseManagerTbService>();
             foreach (var caseManagerTbService in userCaseManagerTbServices)
             {
                 if (!adCaseManagerTbServices.Any(c => caseManagerTbService.Equals(c)))
                 {
-                    caseManagerTbServicesToRemove.Add(caseManagerTbService);
+                    _context.Remove(caseManagerTbService);
                 }
             }
-            foreach (var caseManagerTbService in caseManagerTbServicesToRemove)
-            {
-                userCaseManagerTbServices.Remove(caseManagerTbService);
-            }

```

Comment:
And this was the bug, the way we were removing case manager TB service relations was not enough to persuade EF to delete these relations